src/main/scala/ai/starlake/job/sink/bigquery/BigQuerySparkJob.scala

```diff
@@ -167,38 +167,37 @@ class BigQuerySparkJob(
               finalEmptyDF.options(connectorOptions).save()
             }
           }
-          cliConfig.partitionsToUpdate match {
-            case Nil => // No optimisation requested. This happens when there is no existing dataset
-              sourceDF.write
+
+          partitions.foreach { partitionStr =>
```

Comment:
When dynamic is specified, we need to overwrite each paritition one by one otherwise BQ would scrap the whole table